src/FluffySpoon.AspNet.LetsEncrypt.Azure/AzureAppServiceSslBindingCertificatePersistenceStrategy.cs

```diff
@@ -183,24 +183,50 @@ await client.WebApps.Manager
 
 				foreach (var domain in domainsToUpgrade)
 				{
-					logger.LogDebug("Updating host name bindings for domain {0}", domain);
+					logger.LogDebug("Checking host name binding for domain {0}", domain);
 
 					if (azureOptions.Slot == null)
 					{
-						await client.WebApps.Inner.CreateOrUpdateHostNameBindingWithHttpMessagesAsync(
-							azureOptions.ResourceGroupName,
+						var existingBinding = await client.WebApps.Inner.GetHostNameBindingAsync(azureOptions.ResourceGroupName,
 							appTuple.App.Name,
-							domain,
-							new HostNameBindingInner(
-								azureResourceType: AzureResourceType.Website,
-								hostNameType: HostNameType.Verified,
-								customHostNameDnsRecordType: CustomHostNameDnsRecordType.CName,
-								sslState: SslState.SniEnabled,
-								thumbprint: azureCertificate.Thumbprint));
+							domain);
+
+						if (
+							existingBinding == null ||
+							existingBinding.SslState != SslState.SniEnabled ||
+							existingBinding.Thumbprint != azureCertificate.Thumbprint
```

Comment:
There is some duplication here and below. Can you extract this into a function with a name explaining what it does?

Reply:
How about `DoesBindingNeedUpdating`?

Reply:
Ok, I'll implement that.

Reply:
Done.